src/parser/token.rs

```diff
@@ -180,17 +189,45 @@ impl Display for Token {
 }
 
 impl Token {
-    pub fn new(id: TokenType, val: String) -> Self {
-        Self { id, val }
+    pub fn new(id: StorageType, val: String) -> Self {
+        Self {
+            id: TokenType(id),
+            val,
+        }
     }
 
-    pub fn is_aggregator_with_param(&self) -> bool {
-        is_aggregator_with_param(self.id)
+    pub fn id(&self) -> StorageType {
+        self.id.id()
     }
 }
 
-pub fn is_aggregator_with_param(id: TokenType) -> bool {
-    id == T_TOPK || id == T_BOTTOMK || id == T_COUNT_VALUES || id == T_QUANTILE
+impl TokenType {
+    pub fn new(id: StorageType) -> Self {
+        Self(id)
+    }
+    pub fn id(&self) -> StorageType {
+        self.0
+    }
+    pub fn is_aggregator_with_param(&self) -> bool {
+        self.0 == T_TOPK || self.0 == T_BOTTOMK || self.0 == T_COUNT_VALUES || self.0 == T_QUANTILE
```

Comment:
Is it possible to change this to matches?
```suggestion
        matches!(self.0, T_TOPK | T_BOTTOMK | T_COUNT_VALUES | T_QUANTILE)
```

Reply:
good idea